cmd/igod/main.go

```diff
@@ -1,17 +1,46 @@
 package main
 
 import (
+	"flag"
+	"log"
 	"net/http"
+	"os"
 
+	"github.com/BurntSushi/toml"
 	"github.com/beeceej/iGo/igo/igod"
 	"github.com/beeceej/iGo/igo/interpreter"
 )
 
+type configuration struct {
+	Port string `toml:port`
+}
+
 func main() {
 	i := interpreter.Interpreter{}
 	s := &igod.Server{
 		ServeMux:    http.NewServeMux(),
 		Interpreter: i,
 	}
-	s.Run()
+	port := getPort()
+
+	s.Run(port)
+}
+
+//getPort returns the port based on the server configuration
+func getPort() string {
+	portPtr := flag.String("port", "", "port on which server should run")
+	flag.Parse()
+	var port string
+	if *portPtr != "" {
+		port = *portPtr
+	} else if os.Getenv("IGO_PORT") != "" {
+		port = os.Getenv("IGO_PORT")
+	} else {
+		var configuration configuration
+		if _, err := toml.DecodeFile(".config/igo/config.toml", &configuration); err != nil {
```

Comment:
I would use `filepath.Join(os.Getenv("HOME"), ".config", "igo", "config.toml")`.

the location for this file would be in the users home directory which is why I would recommend looking up the users home directory via env var.